src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -1231,6 +1231,10 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 						if (connectionMngInfo.serverInfo) {
 							connection.options.isCloud = connectionMngInfo.serverInfo.isCloud;
 						}
+						else if (connectionMngInfo.ownerUri.indexOf('|user:|') !== -1) {
+							this._logService.info(`Server info provided is missing the required username for ${uri}`);
+							errorMessage = nls.localize('connectionManagementService.noUsernameForString', "Connection string is missing the required \"User Id\" property.");
```

Comment:
in the context here, there is nothing indicating a connection string, you could simply indicate that the username is missing. also, connection provider could provide a different display name for username field, please use the capabilitiesService to get the actual display name for username property.